### PR TITLE
Add kustomize edit add/remove transformer command #3053

### PR DESCRIPTION
**Pull request by mikeyrcamp**
Ref. #3053 
Similar to edit add/remove patch these commandsadd the ability to add or remove a transformer file path from the kustomization.yaml.

/sig cli
/cc @monopole 
/kind feature

### CODE REVIEW

**Comment by mikeyrcamp** [comment]
check cla

File: kustomize/internal/commands/edit/add/addtransformer.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	transformerFilePaths []string
+}
+
+// newCmdAddTransformer adds the name of a file containing a transformer to the kustomization file.
+func newCmdAddTransformer(fSys filesys.FileSystem) *cobra.Command {
+	var o addTransformerOptions
+
+	cmd := &cobra.Command{
+		Use:   "transformer",
+		Short: "Add the name of a file containing a transformer to the kustomization file.",
```
Comment (monopole) [inline kustomize/internal/commands/edit/add/addtransformer.go:26]: s/transformer/transformer configuration/

File: kustomize/internal/commands/edit/remove/removetransformer_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			fSys := filesys.MakeFsInMemory()
+			testutils_test.WriteTestKustomizationWith(
+				fSys,
+				[]byte(fmt.Sprintf(
+					"transformers:\n  - %s", strings.Join(tc.given.transformers, "\n  - "))))
```
Comment (monopole) [inline kustomize/internal/commands/edit/remove/removetransformer_test.go:132]: ditto on the longer lines here and below, thanks

**Review by monopole — COMMENTED** [review]
some minor nits below, thanks!

File: kustomize/internal/commands/edit/add/addtransformer.go (hunk 1 above)
Comment (mikeyrcamp) [reply kustomize/internal/commands/edit/add/addtransformer.go]: Fixed this and the example in all.go

File: kustomize/internal/commands/edit/remove/removetransformer_test.go (hunk 2 above)
Comment (mikeyrcamp) [reply kustomize/internal/commands/edit/remove/removetransformer_test.go]: Done and fixed in a few other spots.

**Comment by monopole** [comment]
impressive linter.

suggest moving the `given` and `expected` structs, and the `testCases` map out of the Test bodies to one place, and rename the fields from `given.resources` (and `given.transformers`) to `given.items`, and then use the same code in both tests (they do the same thing anyway).  This means adding removeresource_test.go to this PR, but that's fine.

Suggest just changing the package of the test from `remove` to `remove_test` (so that the production package isn't polluted).
Make stuff public if need be.

**Comment by mikeyrcamp** [comment]
@monopole I refactored the test code, and implemented the your suggestions. I'm happy to rework anything again if needed, thanks for your time!

**Pull request merged**